### ISSUE TITLE
Cascade Issues

### ISSUE THREAD

**Issue by benbarefield** [craftyjs/Crafty#378]
There appears to be a bug in the cascading of rotation to attached entities:
var this = Crafty.e("2D, DOM, SomeSprite").attr({ x: 5, y : 5 });
var that = Crafty.e("2D, DOM, SomeSprite").attr({ x : 10, y : 5 });
this.attach(that);
this.rotation = 10;

This code results in that.rotation being set to undefined. Looking into Crafty, I found that rotation cascading is done through a rotate(e) function, and it sets the rotation property to e.theta, but e.theta is undefined during the cascade (I don't know if its ever defined)

I don't know if this is a design decision or not, but z values are not being cascaded either.
this.z = -5 does not change that.z

**Comment by shikolay** [comment]
you can not use "this" for variable
https://developer.mozilla.org/en-US/docs/JavaScript/Reference/Reserved_Words
Try to use other identifier

**Comment by benbarefield** [comment]
Sorry - "this" was a poor choice of name. Change "this" in the code to "anythingElse" and you'll see the issue.

**Comment by benbarefield** [comment]
Since this, the undefined rotation seems to have been fixed, but two other issues came up, each with easy fixes:
First, The child object would rotate opposite the parent:
In _rotate, a difference between the previous angle and new angle is calculated, but its done backwards. I had to change:

``` javascript
var difference = this._rotation - v,
```

to

``` javascript
var difference = v - this._rotation,
```

Second, The polygon used for collision detection was also rotating opposite its parent, even with the above change. In polygon.rotate, I changed:

``` javascript
x = e.o.x + (current[0] - e.o.x) * e.cos + (current[1] - e.o.y) * e.sin;
y = e.o.y - (current[0] - e.o.x) * e.sin + (current[1] - e.o.y) * e.cos;
```

to:

``` javascript
x = e.o.x + (current[0] - e.o.x) * e.cos - (current[1] - e.o.y) * e.sin;
y = e.o.y + (current[0] - e.o.x) * e.sin + (current[1] - e.o.y) * e.cos;
```

If you'd prefer I Fork and submit a pull request for this, I can.

**Comment by starwed** [comment]
Fixed by #450
